jsonschema-core.md

```diff
@@ -1714,62 +1714,35 @@ against the schema defined by this keyword.
 
 #### Keywords for Applying Subschemas Conditionally {#conditional}
 
-Three of these keywords work together to implement conditional application of a
-subschema based on the outcome of another subschema. The fourth is a shortcut
-for a specific conditional case.
+These keywords contain subschemas which are conditionally applied based on the
+instance.
 
-`if`, `then`, and `else` MUST NOT interact with each other across subschema
-boundaries. In other words, an `if` in one branch of an `allOf` MUST NOT have an
-impact on a `then` or `else` in another branch.
+##### `if`, `then`, and `else`
 
-There is no default behavior for `if`, `then`, or `else` when they are not
-present. In particular, they MUST NOT be treated as if present with an empty
-schema, and when `if` is not present, both `then` and `else` MUST be entirely
-ignored.
+These three keywords work together to implement conditional application of a
+subschema based on the outcome of another subschema.
 
-##### `if`
+The value for each of these keywords value MUST be a valid JSON Schema.
 
-This keyword's value MUST be a valid JSON Schema.
-
-This validation outcome of this keyword's subschema has no direct effect on the
-overall validation result. Rather, it controls which of the `then` or `else`
-keywords are evaluated.
-
-Instances that successfully validate against this keyword's subschema MUST also
-be valid against the subschema value of the `then` keyword, if present.
-
-Instances that fail to validate against this keyword's subschema MUST also be
-valid against the subschema value of the `else` keyword, if present.
-
-If [annotations](#annotations) are being collected, they are collected from this
-keyword's subschema in the usual way, including when the keyword is present
-without either `then` or `else`.
-
-##### `then`
+The `if` keyword produces an annotation which is the boolean validation result
+of its subschema against the instance. The validation outcome of the `if`
+keyword's subschema MUST NOT directly affect the overall validation result.
```

Comment:
Reading this, remembering [these tests](https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/main/tests/draft2020-12/unevaluatedProperties.json#L449-L615) which make sure annotation collection happens through `if` for the purposes of determining the validation result of `unevaulatedProperties`, I think there's an opportunity to improve this phrasing.

I realise the phrasing is almost identical to what we had, however what was intedned is that not that it "MUST NOT directly affect the overall validation result", but rather that it "MUST NOT be used as an assertion.".

I don't know if that's less or more confusing, in truth.

When I started writing this suggestion, I thought it was obvious, but turned out to be less so. But I still thought it could be improved.

If anyone thinks it doesn't make it clearer, please just resolve the comment.

Reply:
> make sure annotation collection happens through if for the purposes of determining the validation result of unevaulatedProperties

That's why I said "directly affect."  What you've mentioned is an indirect effect.

Reply:
Right. I think I have trouble passing "directly" the way it's intended. Maybe I need more coffee.